include/envoy/http/filter.h

```diff
@@ -436,6 +436,26 @@ class StreamDecoderFilterCallbacks : public virtual StreamFilterCallbacks {
    * @return The socket options to be applied to the upstream request.
    */
   virtual Network::Socket::OptionsSharedPtr getUpstreamSocketOptions() const PURE;
+
+  /**
+   * Schedules a request for a RouteConfiguration update from the management server.
+   * @param route_config_updated_cb callback to be called when the configuration update has been
+   * propagated to the worker thread.
+   */
+  virtual void requestRouteConfigUpdate(std::function<void()> route_config_updated_cb) PURE;
```

Comment:
I don't think this belongs on the filter callbacks interface. The filter should be able to obtain the `RouteConfigProvider` and the rest should be done by the `RouteConfigProvider` interface.

Reply:
Hmm. ```RouteConfigProvider``` isn't easily accessible from a ```StreamDecoderFilterCallbacks``` -- it's a member of an ```ActiveStream```, for which FilterCallbacks acts as a proxy (for most if not all methods of the interface). Complicating this even more, a ```requestRouteConfigUpdate``` call is delegated to a concrete instance of a ```RouteConfigUpdateRequester```, as scoped and regular RDS route config provider require different implementations (noop vs. making a request).

The closest I can get to what you are suggesting would be to extend the filter interface with an ```ActiveStream``` accessor, but that feels wrong as that would pollute the interface with an implementation detail...

Reply:
It might be reasonable to expose some method to access the `RouteConfigProvider` in `StreamDecoderFilterCallbacks`. This seems a bit cleaner than adding a bunch of quite VHDS specific fields to the interface.